web-ng/src/app/services/dialog/dialog.service.spec.ts

```diff
@@ -17,17 +17,49 @@
 import { TestBed } from '@angular/core/testing';
 
 import { DialogService } from './dialog.service';
-import { MatDialogModule } from '@angular/material/dialog';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { ConfirmationDialogComponent } from '../../components/confirmation-dialog/confirmation-dialog.component';
 
 describe('DialogService', () => {
   let service: DialogService;
+  let dialogSpy: jasmine.Spy;
 
   beforeEach(() => {
     TestBed.configureTestingModule({ imports: [MatDialogModule] });
     service = TestBed.inject(DialogService);
+
+    dialogSpy = spyOn(TestBed.inject(MatDialog), 'open');
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should return correct dialog without discard actions', () => {
+    service.openConfirmationDialog('testTitle', 'testMessage');
+
+    expect(dialogSpy).toHaveBeenCalledWith(ConfirmationDialogComponent, {
+      maxWidth: '500px',
```

Comment:
Only check non-constant values  here and throughout (https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html). You can use a partial matcher you showed me today.